bluepymm/prepare_combos/parse_files.py

```diff
@@ -92,6 +92,53 @@ def read_mm_recipe(recipe_filename):
     """Read a BBP builder recipe and return a pandas.DataFrame with all
     possible (layer, m-type, e-type)-combinations.
 
+    Args:
+        recipe_filename(str): filename of recipe (XML/YAML)
+
+    Returns:
+        A pandas.DataFrame with fields "layer", "fullmtype", and "etype".
+    """
+    from pathlib import Path
+
+    if Path(recipe_filename).suffix == '.xml':
+        return read_mm_recipe_xml(recipe_filename)
+    elif Path(recipe_filename).suffix == '.yaml':
+        return read_mm_recipe_yaml(recipe_filename)
+    else:
+        raise Exception('Please provide an .xml or .yaml as recipe file')
+
+
+def read_mm_recipe_yaml(recipe_filename):
+    """Read a BBP builder recipe and return a pandas.DataFrame with all
+    possible (layer, m-type, e-type)-combinations.
+
+    Args:
+        recipe_filename(str): filename of recipe (YAML)
+
+    Returns:
+        A pandas.DataFrame with fields "layer", "fullmtype", and "etype".
+    """
+    import yaml
+
+    with open(recipe_filename, 'r') as f:
+        recipe = yaml.safe_load(f)
+
+    assert recipe['version'] in ('v2.0',)
```

Comment:
maybe an exception can be raised here instead of the assertion

Reply:
I agree

Reply:
I use assertions for code coverage reasons - for full coverage, I'd need to make a fake yaml file w/ the wrong version, which seems excessive.
The assert still raises an exception, and the message can be made informative.  I doubt anyone will try and catch this exception, so whether it's an `Exception` or `AssertionError`, I'm not sure it matters.

Reply:
I see your point. But the assert statement is removed when Python is called with the -O flag. I don't know if a python rpm or deb package is using the -O flag.

Reply:
You have to explictly enable 'optimizations' (ie: call `python -O ....`), and the performance difference b/c of that is pretty small - I've actually never seen anyone do that.  Hence why I'm not generally worried about using asserts.

Reply:
I see. Would a one liner `if not condition: raise Exception()` not be the same in terms of coverage?

Reply:
not for branch coverage: you'd have to explicitly write tests where `condition` is true, and false

Reply:
Let's create a PEP with the `raiseif` statement :))